### ISSUE TITLE
torch.nn.functional.logsigmoid loses a representable float64 second-derivative component

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#195679]
### 🐛 Describe the bug

`torch.nn.functional.logsigmoid` returns `0.0` for a second-order derivative component whose mathematically correct value is finite, normal, and nonzero in `float64`.

The second derivative of `logsigmoid(x)` is `-sigmoid(x) * sigmoid(-x)`, which is an even function. Therefore, the second derivatives at `x` and `-x` should match. At `x = 37.42994775023705`, the high-precision reference rounds to approximately `-5.551115123125775e-17` for both `x` and `-x`. However, PyTorch collapses the positive-side second derivative to `0.0`, while the negative-side value remains nonzero.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch
import torch.nn.functional as F

dtype = torch.float64

def second_derivative(value):
    x = torch.tensor(value, dtype=dtype, requires_grad=True)
    y = F.logsigmoid(x)
    g1, = torch.autograd.grad(y, x, create_graph=True)
    g2, = torch.autograd.grad(g1, x)
    return g2

actual = torch.tensor(
    [
        second_derivative(37.42994775023705).item(),
        second_derivative(-37.42994775023705).item(),
    ],
    dtype=dtype,
)

expected = torch.tensor(
    [
        -5.551115123125775e-17,
        -5.551115123125775e-17,
    ],
    dtype=dtype,
)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor([ 0.0000e+00, -5.5511e-17], dtype=torch.float64)

expected:
tensor([-5.5511e-17, -5.5511e-17], dtype=torch.float64)
```

The discrepancy occurs in the second-order backward pass. The expected positive-side second derivative is a finite, normal, nonzero `float64` value, but PyTorch collapses it to `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93
